examples/synchronization/multi_function/cont_ai_ci_tdms_sync.py

```diff
@@ -0,0 +1,240 @@
+"""Example of logging multiple synchronized tasks to a single TDMS file using a queue.
+
+This example demonstrates how to:
+1. Synchronize multiple DAQmx tasks using a shared clock
+2. Use a producer-consumer pattern with a queue
+3. Log data from multiple tasks to a single TDMS file
+"""
+
+import os
+import queue
+import threading
+import time
+from collections.abc import Sequence
+from concurrent.futures import ThreadPoolExecutor
+from pathlib import Path
+from typing import List, Union
+
+from nitypes.waveform import AnalogWaveform
+from nptdms import ChannelObject, GroupObject, RootObject, TdmsFile, TdmsWriter
+
+import nidaqmx
+from nidaqmx.constants import (
+    AcquisitionType,
+    Edge,
+)
+from nidaqmx.stream_readers import AnalogMultiChannelReader, CounterReader
+
+os.environ["NIDAQMX_ENABLE_WAVEFORM_SUPPORT"] = "1"
+
+# Configuration
+SAMPLE_RATE = 1000
+SAMPLES_PER_CHANNEL = 1000
+TIMEOUT = 10.0
+# Note: This example currently assumes all tasks run on a single synchronized device.
+# To make this example work across multiple devices, export
+# a shared sample clock from one device to the other via a PFI line.
+DEVICE_NAME = "Dev1"
+
+TaskData = Union[
+    List[AnalogWaveform],  # Analog input: list of waveforms
+    AnalogWaveform,  # Counter input: waveform
```

Comment:
This doesn't match what you're putting in the queue. You're adding a `list[list[AnalogWaveform], AnalogWaveform]`. The outer list really ought to be a tuple.